src/middleware/index.ts

```diff
@@ -1,47 +1,61 @@
-import Redis, { RedisOptions } from 'ioredis';
+import Redis from 'ioredis';
 import { parse, validate } from 'graphql';
 import { GraphQLSchema } from 'graphql/type/schema';
 import { Request, Response, NextFunction, RequestHandler } from 'express';
-
 import buildTypeWeightsFromSchema, { defaultTypeWeightsConfig } from '../analysis/buildTypeWeights';
 import setupRateLimiter from './rateLimiterSetup';
 import getQueryTypeComplexity from '../analysis/typeComplexityAnalysis';
-import { RateLimiterOptions, RateLimiterSelection } from '../@types/rateLimit';
-import { TypeWeightConfig } from '../@types/buildTypeWeights';
+import { ExpressMiddlewareConfig, ExpressMiddlewareSet } from '../@types/expressMiddleware';
 
 // FIXME: Will the developer be responsible for first parsing the schema from a file?
 // Can consider accepting a string representing a the filepath to a schema
 // FIXME: Should a 429 status be sent by default or do we allow the user to handle blocked requests?
 
 /**
  * Primary entry point for adding GraphQL Rate Limiting middleware to an Express Server
- * @param {RateLimiterSelection} rateLimiter Specify rate limiting algorithm to be used
- * @param {RateLimiterOptions} options Specify the appropriate options for the selected rateLimiter
  * @param {GraphQLSchema} schema GraphQLSchema object
- * @param {RedisClientOptions} RedisOptions ioredis connection options https://ioredis.readthedocs.io/en/stable/API/#new_Redis
- * @param {TypeWeightConfig} typeWeightConfig Optional type weight configuration for the GraphQL Schema.
- * Defaults to {mutation: 10, object: 1, field: 0, connection: 2}
+ * @param {ExpressMiddlewareConfig} middlewareConfig
+ *  - One parameter to configure redis client, typeWeights and rate limiting parameters
+ *      - Ratelimiter is required in the setup of the middleware. Developers must explicitly specify this
```

Comment:
These comments may appear in the docstring / hover text when a developer is using the package. I think they can be cleaned up a little and be a little more concise. 

`RateLimiter` should be `rateLimiter`
`ioredis` should be `redis`